core/src/main/java/org/infinispan/notifications/cachelistener/CacheNotifierImpl.java

```diff
@@ -1179,4 +1284,15 @@ public void removeListener(Object listener) {
          }
       }
    }
+
+   private void removeListenerFromFilterIndexingServiceProvider(Object listener) {
```

Comment:
It seems we could do this by changing removeListenerInvocation to protected and then overriding it to do this logic?  This way we don't have to loop through all of the annotations.  We could even refactor out the first line to be separate since that would be shared.